src/testrender/shading.cpp

```diff
@@ -491,7 +528,22 @@ struct OrenNayar final : public BSDF, OrenNayarParams {
     }
 
 private:
-    float A, B;
+    static constexpr float constant1_FON = float(0.5 - 2.0 / (3.0 * M_PI));
+    static constexpr float constant2_FON = float(2.0 / 3.0
+                                                 - 28.0 / (15.0 * M_PI));
+
+    float E_FON_analytic(float mu) const
+    {
+        float AF = 1.0f
+                   / (1.0f
+                      + constant1_FON * sigma);  // Fujii model A coefficient
+        float BF = sigma * AF;                   // Fujii model B coefficient
+        float Si = sqrtf(std::max(0.0f, 1.0f - mu * mu));
```

Comment:
Is mu truly bounded to be <=1 given the small discrepancies of vector normalization in float?
I'd say probably not, and if it isn't, you'll get issues.
If suggest doing a 0,1 clamp on mu just to make sure. If Si >1, then G could become negative, which would be bad.

Reply:
`mu` is guaranteed to be >0 here. We check for the >=1 case by clamping the term inside the `sqrt` for `Si`.